### ISSUE TITLE
compiling boutcore

### ISSUE THREAD

**Issue by johnomotani** [boutproject/BOUT-dev#1279]
Since #1197 compiling boutcore does not work for me. My generated boutcore.pyx has the line
```
# distutils: extra_link_args =
```
If I delete that line and `make` again, boutcore compiles successfully.

The output when it fails is
```
  Generating boutcore.pyx
find: ‘/usr/lib/python3.5/dist-packages’: No such file or directory
  Generating setup.py
  Generating helper.cxx
  Generating helper.h
  Generating resolve_enum.pxd
  Generating boutcpp.pxd
Traceback (most recent call last):
  File "setup.py", line 16, in <module>
    language="c++",             # generate C++ code
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 758, in cythonize
    aliases=aliases)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 664, in create_extension_list
    kwds = deps.distutils_info(file, aliases, base).values
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 564, in distutils_info
    return (self.transitive_merge(filename, self.distutils_info0, DistutilsInfo.merge)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 574, in transitive_merge
    node, extract, merge, seen, {}, self.cimported_files)[0]
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 579, in transitive_merge_helper
    deps = extract(node)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 554, in distutils_info0
    info = self.parse_dependencies(filename)[3]
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 438, in parse_dependencies
    return parse_dependencies(source_filename)
  File "/usr/lib/python3/dist-packages/Cython/Utils.py", line 29, in wrapper
    res = cache[args] = f(*args)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 409, in parse_dependencies
    distutils_info = DistutilsInfo(source)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 198, in __init__
    value = parse_list(value)
  File "/usr/lib/python3/dist-packages/Cython/Build/Dependencies.py", line 132, in parse_list
    if s[0] == '[' and s[-1] == ']':
IndexError: string index out of range
Makefile:19: recipe for target 'boutcore.so' failed
make: *** [boutcore.so] Error 1
```

Not sure how to debug this further. Can you help @dschwoerer?

**Comment by d7919** [comment]
My guess is your `LDFLAGS` variable is empty and it's trying to parse an empty string.

**Comment by d7919** [comment]
What's the output of `bout-config --cflags` for you?

**Comment by johnomotani** [comment]
`$ bin/bout-config --libs`
`-L/home/jomotani/pkg/sundials/instdir/lib -L/home/jomotani/pkg/sundials/instdir -L/home/jomotani/pkg/sundials/instdir -lm -L/home/jomotani/soltransport/BOUT-next-clean-debug/lib -lbout++ -lm -lsundials_ida -lsundials_nvecparallel -lsundials_cvode -lsundials_nvecparallel -lsundials_arkode -lsundials_nvecparallel -lfftw3 -L/usr/lib -lnetcdf_c++4 -L/usr/lib/x86_64-linux-gnu/hdf5/serial -lhdf5_hl -lhdf5 -lpthread -lsz -lz -ldl -lm -lblas -llapack -L/home/jomotani/pkg/sundials/instdir/lib -lsundials_ida -lsundials_nvecparallel -L/home/jomotani/pkg/sundials/instdir -lsundials_cvode -lsundials_nvecparallel -L/home/jomotani/pkg/sundials/instdir -lsundials_arkode -lsundials_nvecparallel -L/home/jomotani/soltransport/BOUT-next-clean-debug/lib -lpvode -lpvpre`

**Comment by d7919** [comment]
Yes that will lead to an empty `ldflags` variable. As a complete guess as to something that might work can you try adding something like

~~~~
if test ".${ldflags}" == "."
then
    ldflags=None
fi
~~~~
to L30 of tools/pylib/_boutcore_build/boutcore.pyx.in

**Comment by johnomotani** [comment]
Unfortunately it didn't like that either
```
g++: error: None: No such file or directory
```
I tried to put an 'if' around L47
```
# distutils: extra_link_args = $ldflags
```
but this works
```
cat <<EOF
# distutils: language=c++
# distutils: include_dirs = $inc_dirs
# distutils: libraries = $libs
# distutils: library_dirs = $lib_dirs
# distutils: sources = helper.cxx
# distutils: extra_compile_args = $flags
EOF
if test ".${ldflags}" != "."
then
    echo "# distutils: extra_link_args = $ldflags"
fi
cat <<EOF
```
Guess I'll make a PR with that as a fix...

**Comment by bendudson** [comment]
Fixed by PR #1280
